### PR TITLE
chore: remove validation on file upload

### PR DESCRIPTION
**Pull request by statictype**
removes the check if a wasm field exists in the abi from the file input validation. this is an old feature before the dry run was implemented and is no longer necessary since the failed dry run will prevent instantiation.

### CODE REVIEW

**Comment by peetzweg** [comment]
I guess the problem is that it tries to validate the wasm even if `isWasmRequired` option is set to false. Which throws an error if no wasm is available, I guess?
https://github.com/paritytech/contracts-ui/blob/66a276e314ee5c2da85277b7a3eaed7a9667f9aa/src/ui/hooks/useMetadata.ts#L80

This should be fixed if we just move the validation code from outside the `if` statement into the `if` statement. 

https://github.com/paritytech/contracts-ui/blob/66a276e314ee5c2da85277b7a3eaed7a9667f9aa/src/ui/hooks/useMetadata.ts#L78-L88

File: src/ui/hooks/useMetadata.ts [hunk 1]
```diff
@@ -75,18 +75,6 @@ function validate(metadata: Abi | undefined, { isWasmRequired }: Options): Valid
     };
   }
 
-  const wasm = metadata.info.source.wasm;
-  const isWasmEmpty = wasm.isEmpty;
-  const isWasmInvalid = !WebAssembly.validate(wasm);
-
-  if (isWasmRequired && (isWasmEmpty || isWasmInvalid)) {
```
Comment (peetzweg) [inline src/ui/hooks/useMetadata.ts:82]: If we just go with removal, we should reflect that as well in the `Options`.

https://github.com/paritytech/contracts-ui/blob/66a276e314ee5c2da85277b7a3eaed7a9667f9aa/src/ui/hooks/useMetadata.ts#L11-L13

Comment (statictype) [reply src/ui/hooks/useMetadata.ts:82]: yup, didn't finish this yet. marked as draft now.

Comment (statictype) [reply src/ui/hooks/useMetadata.ts:82]: it's a case of bad naming here. `isWasmRequired` means you are calling `instantiate_with_code`

this PR only affects the file input validation. nothing else needs to be removed actually

**Comment by peetzweg** [comment]
Not sure what the cause for this change is? Is there a ticket describing in the issue? 

Validation is probably not needed as the instantiation extrinsic dryrun would fail? 🤷

**Review by peetzweg — APPROVED** [review]

**Comment by statictype** [comment]
> Validation is probably not needed as the instantiation extrinsic dryrun would fail?

exactly

> Not sure what the cause for this change is? Is there a ticket describing in the issue?

except the fact it's not needed, removing it allows experimentation with other targets like RISC V

**Pull request merged**